### PR TITLE
Fix get_route_for_model for ContentTypes

### PR DESCRIPTION
**Pull request by timizuoebideri1**
# Closes: #3080 
# What's Changed

Running `get_route_for_model(ConentType, api=True...)` would yield `content-types-api:contenttype-list` rather than the newly created endpoint for contenttypes because `ContentType` is not a core model in nautobot (`extras-api:contenttype-list`)

# TODO

- [ ] Explanation of Change(s)
- [ ] Added change log fragment(s) (for more information see [the documentation](https://docs.nautobot.com/projects/core/en/stable/development/#creating-changelog-fragments))
- [ ] Attached Screenshots, Payload Example
- [ ] Unit, Integration Tests
- [ ] Documentation Updates (when adding/changing features)
- [ ] Example Plugin Updates (when adding/changing features)
- [ ] Outline Remaining Work, Constraints from Design

### CODE REVIEW

**Review by bryanculver — APPROVED** [review]
LGTM. Curious to know why we can't use the Django default for ContentType. Doesn't appear we do anything fancy with our ContentType view. Is their default not read-only?

**Comment by glennmatthews** [comment]
> LGTM. Curious to know why we can't use the Django default for ContentType. Doesn't appear we do anything fancy with our ContentType view. Is their default not read-only?

There is no default REST API provided by the `contenttypes` app AFAIK. We provide one for the ContentType model through our `extras` app.

File: nautobot/utilities/utils.py [hunk 1]
```diff
@@ -105,7 +105,13 @@ def get_route_for_model(model, action, api=False):
         model = get_model_from_name(model)
 
     suffix = "" if not api else "-api"
-    prefix = f"{model._meta.app_label}{suffix}:{model._meta.model_name}"
+    # Because ContentType is not a core model in nautobot, running a 'ContentType. meta.app label'
+    # would return "contenttypes" rather than "extras".
```
Comment (glennmatthews) [inline nautobot/utilities/utils.py:109]: Perhaps:

```suggestion
    # The `contenttypes` app doesn't provide REST API endpoints,
    # but Nautobot provides one for the ContentType model in our `extras` app.
```

**Review by glennmatthews — COMMENTED** [review]
1) Do we need a similar fix for the `Group` model and the `users/groups/` REST URL?
2) Would you mind looking into how difficult it would be to fix #3300 in this same PR?

**Comment by bryanculver** [comment]
> > LGTM. Curious to know why we can't use the Django default for ContentType. Doesn't appear we do anything fancy with our ContentType view. Is their default not read-only?
> 
> There is no default REST API provided by the `contenttypes` app AFAIK. We provide one for the ContentType model through our `extras` app.

DOH. Didn't think beyond the `get_route_for_model` being our invention.

**Review by jathanism — APPROVED** [review]

**Comment by timizuoebideri1** [comment]
> * Do we need a similar fix for the `Group` model and the `users/groups/` REST URL?
> * Would you mind looking into how difficult it would be to fix #3300 in this same PR?

1. I suppose we'll need a similar fix for 'Group'. 
2. I have little to no idea how GraphQL works, so it may take longer because I need to learn how it works first.

**Comment by glennmatthews** [comment]
> > * Would you mind looking into how difficult it would be to fix [Unable to query `content_types` via GraphQL #3300](https://github.com/nautobot/nautobot/issues/3300) in this same PR?
>     2. I have little to no idea how GraphQL works, so it may take longer because I need to learn how it works first.

No worries, that can be deferred since it's a separate issue. Thanks!

File: changes/3341.fixed [hunk 2]
```diff
@@ -0,0 +1 @@
+Fix the issue with the ContentType reverse return, i.e get_route_for_model(ContentType, api=True...).
```
Comment (glennmatthews) [inline changes/3341.fixed:1]: Perhaps:

```suggestion
Fixed missing `get_route_for_model()` logic for the `ContentType` and `Group` models.
```

File: nautobot/utilities/tests/test_utils.py [hunk 3]
```diff
@@ -231,6 +233,10 @@ def test_get_route_for_model(self):
         self.assertEqual(get_route_for_model(Device, "list", api=True), "dcim-api:device-list")
         self.assertEqual(get_route_for_model("dcim.site", "detail", api=True), "dcim-api:site-detail")
         self.assertEqual(get_route_for_model(Site, "detail", api=True), "dcim-api:site-detail")
+        self.assertEqual(get_route_for_model(ContentType, "list", api=True), "extras-api:contenttype-list")
+        self.assertEqual(get_route_for_model(ContentType, "detail", api=True), "extras-api:contenttype-detail")
+        self.assertEqual(get_route_for_model(Group, "list", api=True), "users-api:group-list")
+        self.assertEqual(get_route_for_model(Group, "detail", api=True), "users-api:group-detail")
```
Comment (glennmatthews) [inline nautobot/utilities/tests/test_utils.py:239]: Hmm, I suppose for the `api=False` case we don't actually have viable routes for either of these models, do we? I guess it's inline with how `get_route_for_model` currently works that it just tries to guess the route name, but doesn't actually verify that it's valid though. Ah well.

**Review by glennmatthews — APPROVED** [review]

**Pull request merged**